src/sql/src/plan/error.rs

```diff
@@ -644,6 +653,10 @@ impl fmt::Display for PlanError {
             Self::InvalidSinkFrom { name, item_type } => {
                 write!(f, "{name} is a {item_type}, which cannot be exported as a sink")
             },
+            Self::InvalidDependency { name, item_type } => {
+                let a = if *item_type == CatalogItemType::Index { "an" } else { "a" };
+                write!(f, "{name} is {a} {item_type}, which cannot be depended upon")
+            },
```

Comment:
The error message is consistent with `InvalidSinkFrom`, but different than before. I'm not married to this, there isn't much consistency in our error messages anyway. Fine with changing it to the old format, if anyone has strong opinions!